bedita-app/models/objects/content/video.php

```diff
@@ -24,8 +24,101 @@
  */
 class Video extends BeditaStreamModel
 {
-	var $useTable = "videos";
-	var $actsAs = array();
-	public $objectTypesGroups = array("multimedia", "leafs", "related");
+    public $useTable = 'videos';
+
+    public $actsAs = array();
+
+    public $objectTypesGroups = array('multimedia', 'leafs', 'related');
+
+    /**
+     * Load captions in video data.
+     *
+     * @param array $results Fetched videos.
+     * @return array
+     */
+    public function afterFind(array $results)
+    {
+        $results = parent::afterFind($results);
+
+        foreach ($results as &$result) {
+            $result['captions'] = $this->getCaptions($result['id']);
+        }
+        unset($result);
+
+        return $results;
+    }
+
+    /**
+     * Save captions after video has been saved.
+     *
+     * @param bool $created Is this a freshly created entity?
+     * @return void
+     */
+    public function afterSave($created)
+    {
+        parent::afterSave($created);
+
+        if (isset($this->data[$this->alias]['captions'])) {
+            $this->saveCaptions($this->id, $this->data[$this->alias]['captions']);
+        }
+    }
+
+    /**
+     * Return map of captions indexed by their language.
+     *
+     * @param int $videoId Video ID.
+     * @return array
+     */
+    protected function getCaptions($videoId)
+    {
+        $CaptionModel = ClassRegistry::init('Caption');
+        $found = $CaptionModel->find('all', array(
+            'conditions' => array(
+                'object_id' => $videoId,
+            ),
+            'contain' => array('BEObject'),
+        ));
+
+        return Set::combine($found, '{n}.lang', '{n}');
+    }
+
+    /**
+     * Save captions for a video.
+     *
+     * @param int $videoId Video ID.
+     * @param array $data Map of lang to VTT contents.
+     * @return void
+     */
+    protected function saveCaptions($videoId, array $data)
+    {
+        $data = array_filter(
+            $data,
+            function ($datum) {
+                return !empty($datum['description']);
+            }
+        );
+
+        $CaptionModel = ClassRegistry::init('Caption');
+
+        foreach ($data as $datum) {
+            $datum['object_id'] = $videoId;
+            $CaptionModel->save($datum);
+        }
+
```

Comment:
The `unset($result)` is put there for "safety" reasons. Since the foreach loop iterates using a symbol passed by reference `&$result`, if we used `$result` for something else _after_ the loop, this would affect the last element in the iteration:

```php
$arr = [1, 2, 3];
foreach ($arr as &$item) {
  $item = $item * 2;
}
// Now: $arr === [2, 4, 6]

$item = 'hello world';
// Now: $arr === [2, 4, 'hello world'] 😱
```

The `unset($result)` "looses" the reference and makes it safe to re-use the symbol, so I usually add it every time I do a `foreach ($arr as &$item)` loop so that I won't get burnt in the future should I make changes to the same part of code. In this case, since `$datum` ain't a reference, it shouldn't be necessary. But it's perfectly OK to add it anyway if you wish.

Reply:
ok, not necessary. thanks for the detailed explanation ;)